Cod3rsGrowth.Web/wwwroot/Webapp/Controller/AdicionarVenda.controller.js

```diff
@@ -0,0 +1,140 @@
+﻿sap.ui.define([
+    "ui5/carro/controller/BaseController",
+    "sap/ui/core/routing/History",
+    "sap/ui/model/json/JSONModel",
+    "ui5/carro/model/formatter",
+    "sap/m/MessageToast",
+    "ui5/carro/model/validacao"
+
+], function (BaseController, History, JSONModel, Formatter, MessageToast, validacao) {
+    "use strict";
+
+    var ModeloCarro = "Carros"
+    var idDoInputCpf = "InputCpf"
+    var idDoInputPago = "InputPago"
+    var idDoInputNome = "InputNome"
+    var idDoInputEmail = "InputEmail"
+    var idDoInputTelefone = "InputTelefone"
+    var urlCarro = "http://localhost:5071/api/Carros/Disponiveis"
+
+
+    return BaseController.extend("ui5.carro.controller.AdicionarVenda", {
+        formatter: Formatter,
+
+        onInit() {
+            this._carregarCarros();
```

Comment:
Quando a rota identificada por ROTA_LISTAGEM é ativada, o método _carregarReservas é chamado, permitindo que você execute lógica específica, como carregar dados ou atualizar a interface do usuário.

Cria essa função e usa no seu oninit em vez de usar o metodo _carregarCarros diretamente.

aoCoincidirRota: function () {
    this.processarEvento(() => {
        var oRouter = sap.ui.core.UIComponent.getRouterFor(this);
        oRouter.getRoute(ROTA_LISTAGEM).attachMatched(this._carregarReservas, this);
    });
},